### ISSUE TITLE
Question: Java 17 and Spring Boot 3 branch

### ISSUE THREAD

**Issue by kwonglau** [eventuate-foundation/eventuate-common#129]
Hi Chris,

Thanks for creating the eventuate framework, I like it a lot.

I would like to offer some help to move the project forward for some dependencies upgrade such as junit 5 (eventuate-util), java 17 as baseline and spring boot 3 (auto-configuration is not working currently in SB3 because the file has been moved to a new location and with a new name, https://github.com/spring-projects/spring-boot/issues/32566). However. I have some pain points to create pr when micronaut is in the same code base in eventuate-common and eventuate-kafka (v 3.3.1 without using zookeeper). It is very hard to get all the test pass and give that the version it is using in the project is one major version behind. It takes time for one to learn about the framework when other only want to contribute for a single framework support. 

Are there any plan to extract micronaut to a different repo? Any thought ?

Thanks,
Anthony

**Comment by cer** [comment]
A few thoughts:

First, I would definitely appreciate your contributions. Please sign a [CLA](https://chrisrichardson.net/legal/)

Second, upgrading various dependencies is definitely a good thing. However, we need to be careful about requiring them since folks might still be using older versions. 

For example, Eventuate should support

* the oldest version of Spring Boot that is [still supported (non-commercially)](https://spring.io/projects/spring-boot#support), e.g. currently 2.4
* all later versions, including Spring Boot 3 

There are potentially similar concerns about Kafka since folks aren't always on the latest version.

Third, I would be beneficial to move Micronaut code into separate projects and upgrade it although that is deferring work that needs to be done eventually.

**Comment by kwonglau** [comment]
thanks

**Comment by kwonglau** [comment]
will you consider raising the java baseline to java 17 anytime soon?

e.g. after 2023-11-24, the eventuate codebase will require minimum java 17 and spring boot 3.1.x. unless they are a paid support eventuate client, they will get update back port to spring boot 2?

**Comment by cer** [comment]
Yes. I think good policy is for Eventuate to be based on the oldest (non-commercial) supported version of Spring, e.g. 2.7 as of today.

**Comment by ddaeyongee** [comment]
> Yes. I think good policy is for Eventuate to be based on the oldest (non-commercial) supported version of Spring, e.g. 2.7 as of today.

@cer 
Hello, this is Taeyong (ddaeyongee) of LG CNS.
Thanks to your hard work last year, we were able to successfully open the project. Thank you again.

This is a question about the answer above. We are making good use of the non-commercial spring 2.7 based Eventuate CDC module. The year has changed, and do you have any concerns about the spring 3.x version? Do you still feel the same way?

I'll wait for the answer to your question.
New Year's greetings are late. happy new year! Sir!

**Comment by cer** [comment]
> New Year's greetings are late. happy new year! Sir!

Hi. Sorry for the late reply. I'm pleased to hear that things are going well.

My plan is to upgrade Eventuate platform (libraries + CDC) to use Spring Boot 3.x and other up to date dependencies.
